client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ListenerCacheManagerStopTest.java

```diff
@@ -0,0 +1,118 @@
+package org.infinispan.client.hotrod.event;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.test.HotRodClientTestingUtil;
+import org.infinispan.commons.util.Util;
+import org.infinispan.server.hotrod.HotRodServer;
+import org.infinispan.server.hotrod.test.HotRodTestingUtil;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.net.ServerSocket;
+
+import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+import static org.infinispan.test.TestingUtil.extractField;
+import static org.infinispan.test.fwk.TestCacheManagerFactory.createCacheManager;
+
+/**
+ * Test to verify that the listener threads are stopped.
+ *
+ * @author gustavonalle
+ * @since 7.2
+ */
+@Test(groups = "functional", testName = "client.hotrod.event.ListenerStopTest")
+public class ListenerCacheManagerStopTest extends AbstractInfinispanTest {
+
+   HotRodServer hotRodServer;
+   RemoteCacheManager remoteCacheManager;
+   RemoteCache<Integer, String> cache;
+
+   @BeforeMethod
+   public void setup() throws IOException {
+      Integer port = getRandomFreePort();
+      hotRodServer = HotRodTestingUtil.startHotRodServer(createCacheManager(hotRodCacheConfiguration()), port);
+      remoteCacheManager = new RemoteCacheManager(
+            new ConfigurationBuilder()
+                  .addServer()
+                  .port(port)
+                  .host(HotRodTestingUtil.host())
+                  .connectionTimeout(3000)
+                  .socketTimeout(3000)
+                  .build()
+      );
+      cache = remoteCacheManager.getCache();
+   }
+
+   @AfterMethod
+   public void tearDown() {
```

Comment:
Cache manager is being leaked here. HR server won't stop cache manager...

To avoid such leaks, it's better to extend existing classes, e.g. `SingleHotRodServerTest`